src/3_branch.js

```diff
@@ -1123,27 +1181,52 @@ if (config.WEB_BUILD) {
  *     }
  * });
  * ```
+ * ___
+ *
+ * ### closeBanner()
+ *
+ * #### Closing the App Banner Programmatically
+ *
+ * The App Banner includes a close button the user can click, but you may want to close the banner with a timeout, or via some
+ * other user interaction with your web app. In this case, closing the banner is very simple by calling `Branch.closeBanner()`.
+ *
+ * ##### Usage
+ * ```js
+ * branch.closeBanner();
+ * ```
+ *
  */
 	Branch.prototype['banner'] = function(options, linkData) {
-		var bannerOptions = {
-			icon: options['icon'] || '',
-			title: options['title'] || '',
-			description: options['description'] || '',
-			openAppButtonText: options['openAppButtonText'] || 'View in app',
-			downloadAppButtonText: options['downloadAppButtonText'] || 'Download App',
-			iframe: typeof options['iframe'] == 'undefined' ? true : options['iframe'],
-			showiOS: typeof options['showiOS'] == 'undefined' ? true : options['showiOS'],
-			showAndroid: typeof options['showAndroid'] == 'undefined' ? true : options['showAndroid'],
-			showDesktop: typeof options['showDesktop'] == 'undefined' ? true : options['showDesktop'],
-			disableHide: !!options['disableHide'],
-			forgetHide: !!options['forgetHide'],
-			makeNewLink: !!options['make_new_link']
-		};
+		var self = this;
+		this._queue(function(next) {
+			var bannerOptions = {
+				icon: options['icon'] || '',
+				title: options['title'] || '',
+				description: options['description'] || '',
+				openAppButtonText: options['openAppButtonText'] || 'View in app',
+				downloadAppButtonText: options['downloadAppButtonText'] || 'Download App',
+				iframe: typeof options['iframe'] == 'undefined' ? true : options['iframe'],
+				showiOS: typeof options['showiOS'] == 'undefined' ? true : options['showiOS'],
+				showAndroid: typeof options['showAndroid'] == 'undefined' ? true : options['showAndroid'],
+				showDesktop: typeof options['showDesktop'] == 'undefined' ? true : options['showDesktop'],
+				disableHide: !!options['disableHide'],
+				forgetHide: !!options['forgetHide'],
+				make_new_link: !!options['make_new_link']
+			};
 
-		if (typeof options['showMobile'] != 'undefined') {
-			bannerOptions.showiOS = bannerOptions.showAndroid = options['showMobile'];
-		}
+			if (typeof options['showMobile'] != 'undefined') {
+				bannerOptions.showiOS = bannerOptions.showAndroid = options['showMobile'];
+			}
 
-		banner(this, bannerOptions, linkData, this._storage);
+			banner(self, bannerOptions, linkData, self._storage);
+			next();
+		});
+	};
+
+	Branch.prototype['closeBanner'] = function() {
```

Comment:
Agh, you're correct on number 1. I had originally removed the nonInit error throws,  and at that point, the queue did work with every function. When I put them back in, I didn't think to put them back inside the queue.
With that said, yes, all of this makes sense and it sounds like a great idea. I'll get to work on it.